cps_stage4/extrapolation.py

```diff
@@ -205,85 +212,92 @@ def _read_data(self, growth_rates, cps_benefit, cps_weights,
 
-        cps_benefit.set_index("SEQUENCE", inplace=True)
+        cps_benefit.set_index('SEQUENCE', inplace=True)
 
         self.WT = cps_weights
         growth_rates = pd.read_csv(growth_rates, index_col=0)
         for benefit in benefit_names:
 
             # create benefit targets
             benefit_2014 = (cps_benefit[benefit] * cps_benefit.WT).sum()
-            benefit_targets = benefit_2014 * (1 + growth_rates['{0}_benefit_growth'
-                                                               .format(benefit)])
+            benefit_targets = (benefit_2014 *
+                               (1 + growth_rates['{}_benefit_growth'
+                                                 .format(benefit)]))
             benefit_targets = benefit_targets[1:]
 
             # extract program benefit in 2014 to base_benefits array
-            base_benefits_col = [col for col in list(cps_benefit) if
-                                 col.startswith('{0}_VAL'.format(benefit.upper()))]
+            base_benefits_col = [col for col in list(cps_benefit)
+                                 if col.startswith(
+                                         '{}_VAL'.format(benefit.upper())
+                                 )]
             base_benefits = cps_benefit[base_benefits_col]
             base_benefits.columns = list(range(len(base_benefits.columns)))
-            # Create Participation targets from tax-unit individual level markers
-            # and growth rates from SSA
-            base_participation = pd.DataFrame(np.where(base_benefits > 0, 1, 0),
+            # Create Participation targets from tax-unit individual
+            # level markers and growth rates from SSA
+            col_list = list(range(len(base_benefits.columns)))
+            base_participation = pd.DataFrame(np.where(base_benefits > 0,
+                                                       1, 0),
                                               index=self.index,
-                                              columns=list(range(len(base_benefits.columns))))
-
+                                              columns=col_list)
 
             # print(benefit,'value_counts baseline')
             # print(base_participation.sum(axis=1).value_counts())
             total_particpants = (base_participation.sum(axis=1) *
                                  cps_benefit.WT).sum()
             participant_targets = total_particpants * \
-                (1 + growth_rates['{0}_participation_growth'.format(benefit)])
-            # extract probability of participation in program from tax-unit database
+                (1 + growth_rates['{}_participation_growth'.format(benefit)])
+            # extract probability of participation in program from
+            # tax-unit database
             prob_col = [col for col in list(cps_benefit)
-                        if col.startswith('{0}_PROB'.format(benefit.upper()))]
+                        if col.startswith('{}_PROB'.format(benefit.upper()))]
```

Comment:
@martinholmer what's the benefit of using `'{}'.format(...)` over `'{0}'.format(...)`

Reply:
It's one character less.

Reply:
OK, thanks for the explanation.